server/service/queries.go

```diff
@@ -518,11 +537,30 @@ func (svc *Service) ApplyQuerySpecs(ctx context.Context, specs []*fleet.QuerySpe
 		}
 	}
 	// 3. Apply the queries.
+
+	// first, find out if we should delete query results
+	queriesToDiscardResults := make(map[uint]bool)
+	for _, query := range queries {
+		dbQuery, err := svc.ds.QueryByName(ctx, query.TeamID, query.Name)
+		if err != nil && !fleet.IsNotFound(err) {
+			return ctxerr.Wrap(ctx, err, "fetching saved query")
+		}
+
+		if dbQuery == nil {
+			// then we're creating a new query, so move on.
+			continue
+		}
+
+		if query.DiscardData || query.Logging != fleet.LoggingSnapshot || query.Query != dbQuery.Query {
```

Comment:
Similar here for `DiscardData` and `Logging`: https://github.com/fleetdm/fleet/pull/14302/files#r1350514637